messaging/src/main/java/org/axonframework/common/ProcessUtils.java

```diff
@@ -49,4 +52,33 @@ public static void executeWithRetry(Runnable runnable, Predicate<RuntimeExceptio
 
         throw lastException;
     }
+
+    /**
+     * Executes an action, with potential retry in case the result is false. Exception handling should be taken care of
+     * within the action if needed.
+     *
+     * @param runnable      action to execute
+     * @param retryInterval time to wait between retries of the action
+     */
+    public static void executeUntilTrue(BooleanSupplier runnable, long retryInterval) {
+        boolean result = runnable.getAsBoolean();
+        while (!result) {
+            try {
+                Thread.sleep(retryInterval);
+                result = runnable.getAsBoolean();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    public static void await(Future<?> future){
```

Comment:
I think we would benefit from an `await` that takes a timeout. For use in the Coordinator, actually. This timeout can then be provided to the `CompletableFuture#get` method.